community/kernel/src/main/java/org/neo4j/kernel/api/proc/Procedure.java

```diff
@@ -36,7 +37,7 @@
 public interface Procedure
 {
     ProcedureSignature signature();
-    Stream<Object[]> apply( Context ctx, Object[] input ) throws ProcedureException;
+    RawIterator<Object[], ProcedureException> apply( Context ctx, Object[] input ) throws ProcedureException;
```

Comment:
To have Procedure<T> as a procedure that will return result of type T and then have RawIterator<T, ProcedureException> apply method. This looks fine for newly introduced basic provided procedures.